tools/booster/Proxy.hs

```diff
@@ -295,8 +343,13 @@ respondEither mbStatsVar booster kore req = case req of
                         | null filteredNexts ->
                             -- HACK. Would want to return the prior state
                             res{reason = Stuck, nextStates = Nothing}
-                    _otherReason ->
-                        res{state = simplifiedState, nextStates}
+                    _otherReason
+                        | null filteredNexts ->
+                            res{state = simplifiedState, nextStates = Nothing}
+                        | otherwise ->
+                            res{state = simplifiedState, nextStates = Just filteredNexts}
```

Comment:
It turns out, we were not returning the simplified states, but rather the ones before simplification. Now here comes the fun part:
* returning `filteredNexts` in `Branching`, as originally intended, breaks an integration tests, specifically https://github.com/runtimeverification/hs-backend-booster/blob/main/test/rpc-integration/test-no-evaluator/
* The test brakes because `#Not (#Exists X. X #Equals 12)` is simplified into  `#Not (X #Equals 12)` (provided I understand the kore json corectly).

I will isolate this change into a separate PR.